src/app/shared-services/diary/diary.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@angular/core'
+import { v4 as uuid } from 'uuid'
+import { DiaryEntry } from 'src/app/model/diary-entry'
+import { SqliteService } from '../db/sqlite.service'
+import { Platform } from '@ionic/angular'
+import {
+  FilesystemDirectory,
+  FilesystemEncoding,
+  Plugins,
+} from '@capacitor/core'
+
+const { Filesystem, Share } = Plugins
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DiaryService {
+  constructor(private dbService: SqliteService, private platform: Platform) {}
+
+  async getDiary(): Promise<DiaryEntry[]> {
+    return this.dbService.getDiary()
+  }
+
+  async getDiaryEntry(id: string): Promise<DiaryEntry> {
+    return this.dbService.getDiaryEntry(id)
+  }
+
+  async createDiaryEntry(date: Date, content: string): Promise<void> {
+    const entry = new DiaryEntry(uuid(), new Date(), new Date(), date, content)
+    return this.dbService.upsertDiaryEntry(entry)
+  }
+
+  async updateDiaryEntry(
+    id: string,
+    date: Date,
+    content: string
+  ): Promise<void> {
+    const entry = await this.dbService.getDiaryEntry(id)
+    const update = new DiaryEntry(
+      entry.id,
+      entry.created,
+      new Date(),
+      date,
+      content
+    )
+    return this.dbService.upsertDiaryEntry(update)
+  }
+
+  async deleteDiaryEntry(id: string) {
+    return this.dbService.deleteDiaryEntry(id)
+  }
+
+  async exportDiary() {
+    try {
+      const diary = await this.getDiary()
+      const fileData = diary
+        .map((d) => {
+          return `\n${d.date.toISOString()}\n${d.content}\n`
+        })
+        .join(`\n===========\n`)
```

Comment:
What purpose does this joined string serve?

Reply:
I used the join in order to have a more readable .txt file

Example:
```

2021-12-23T13:16:18.000Z
Hello World

===========

2021-12-23T13:16:30.000Z
this is my second diary entry
```